improver_tests/acceptance/test_checksums.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2020 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Bulk checking and updating of checksum file"""
+
+import os
+import pathlib
+
+import pytest
+
+from . import acceptance as acc
+
+pytestmark = [pytest.mark.acc, acc.skip_if_kgo_missing]
+
+
+def test_kgo_checksums():
```

Comment:
Is it correct that this test is a bit different from other tests in `improver_tests/acceptance/test_checksums.py` for a few reasons?

- It's not testing a CLI
- We need to run this explicitly, if we want to use the `RECREATE_KGO` functionality, so that the checksum file can be rebuilt.

Does it matter if this test is run in parallel with changes to other files?

Reply:
Yes, this test is a bit different from others in the acceptance directory. This is the best place that I could think of to put the test, but I'm open to alternative suggestions for where it could go.

There are two options for rebuilding the checksum file:
- Set the `RECREATE_KGO` environment variable and run this test. Running this one test case by itself is OK - don't need to run the whole test suite.
- Calculate the checksum through another method (eg. `sha256sum` command line tool) and edit `SHA256SUMS` with a text editor.

The worst case scenario for doing an update is: run `test_kgo_checksums` to add new input files, run of all acceptance tests to generate new outputs, run `test_kgo_checksums` to add new output files. In practice, much more targeted use should be possible.

Each pytest process will read the checksum file once and use those cached in-memory values for all the rest of the tests, rather than re-reading the checksum file for each test. There may be multiple parallel pytest processes if using pytest-xdist.
I think the arrangement here is safe to run in parallel, and I've used pytest-xdist often to speed test suite runs up when working on this PR with no problems encountered.